src/middleware/tests/overload-protection-middleware.test.ts

```diff
@@ -0,0 +1,19 @@
+import { expect } from "chai";
+import { describe } from "mocha";
+
+import { applyOverloadProtection } from "../overload-protection-middleware";
+
+describe("applyOverloadProtection", () => {
+  // Sinon doesn't work well with overloadProtection hence just checking if the return exists
+  it("should call overloadProtection with correct options in production mode", () => {
+    const isProduction = true;
+    const result = applyOverloadProtection(isProduction);
+    expect(result).to.exist;
+  });
+
+  it("should call overloadProtection with correct options in non-production mode", () => {
+    const isProduction = false;
+    const result = applyOverloadProtection(isProduction);
+    expect(result).to.exist;
+  });
+});
```

Comment:
I think we're just asserting the middleware is returning a value, but not that it'd doing anything of value for overload protection. Happy to help figure out how to assert this with sinon